openassessment/xblock/job_sample_grader/code_grader.py

```diff
@@ -14,7 +14,46 @@
 OOP_PROBLEM_NAMES = ["design-problem-1", "design-problem-2", "design-problem-3"]
 
 
-class TestGrader:
+class CodeGraderUtils:
+    """
+    A class to hold common methods for code grading operations.
+    """
+
+    @staticmethod
+    def is_design_problem(problem_name):
+        """
+        Temporary helper method to check if a coding problem is a design problem.
+        """
+        return problem_name in OOP_PROBLEM_NAMES
+
+    @staticmethod
+    def get_error_response(run_type, error):
+        """
+        Create a sample error response for a given run and the error to be displayed.
+        """
+        return {
+            'run_type': run_type,
+            'total_tests': 0,
+            'correct': 0,
+            'incorrect': 0,
+            'output': None,
+            'error': [error]
+        }
+
+    @staticmethod
+    def truncate_error_output(output):
+        """
+        Truncate error output to last 150 lines if it is very long.
+        """
+        if len(output.split('\n')) > 150:
+            actual_output = output.split("\n")[-150:]
+            actual_output.append("... Extra output Trimmed.")
+            return "\n".join(actual_output)
+        return output
```

Comment:
create a new utils.py along with code_grader.py and move these methods inside that file. Keep them separate instead of creating CodeGraderUtils class(this is because the code will be moved to job_sample_grader/utils.py file and it is obvious the utils are for code grader only). The object pattern should be used when things connect logically. utility methods are different methods used by a module for its different operations.